File: api/src/main/java/com/trabalho/api/dto/UsuarioDTO.java
package com.trabalho.api.dto;

import java.util.List;

import com.trabalho.api.model.Permissoes;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class UsuarioDTO {
    protected Long id;
    protected String nome;
    protected String email;
    protected boolean ativo;
    protected EnderecoDTO endereco;
    protected List<Permissoes> permissoes;
}

File: cliente-pooweb2/src/app/app-core/dto/UsuarioDto.ts
import { EnderecoDto } from "./EnderecoDto";

export class UsuarioDto{
    id: number;
    nome: string;
    email: string;
    ativo: boolean;
    endereco: EnderecoDto;
    permissoes: Permissao[];

    constructor(){}
}

export enum Permissao{
    ADMIN_SISTEMA = "ADMIN_SISTEMA",
    ADMIN_EMPRESA = "ADMIN_EMPRESA",
    ADMIN_ESTABELECIMENTO = "ADMIN_ESTABELECIMENTO",
    CLIENTE = "CLIENTE"
}